contributed_definitions/NXmpes.nxdl.xml

```diff
@@ -314,7 +314,7 @@
                      annealing).
                 </doc>
             </field>
-            <group name="preparation_description" type="NXnote">
+            <group name="preparation_description" type="NXnote" optional="true">
```

Comment:
Commented on the fairmat-proposal page by Anders Frisk (Scienta Omicron) and Marek Matas (Thermofischer), that the fields `preparation_description`, `temperature` (recommended), `situation` and `gas_pressure` should be optional or recommended instead of required. The rationale is that this information is not always available or useful in the measurement context.

Reply:
In general, I we should reconsider a lot of required fields in NXmpes. There are two ways to look at this. One would be the viewpoint of an "application", and what it requires. This would be really pretty limited for NXmpes, where it is even not that clear what this application should be. The other viewpoint that we partly followed with the original NXmpes was to require a FAIR description of data. This is where most requirements like e.g. user, but also sample description etc. came from. We should probably back down on this if we want a more widely accepted definition, and set those to recommended.

Reply:
We could also use the recommended as an additional layer to imply a more rigid representation of the data according to FAIRness. Currently, it is not always clear to me when to use optional and when to use recommended. We could say that all required fields request the minimal accepted definition and all recommended fields yield a definition which we would consider a full FAIR dataset. Maybe we could even think of enhancing this to levels of FAIRness, e.g., we associate a required/recommended level with a field and this yields a file according to certain level of fairness (which is just a generalization of having the three levels of required, recommended and optional).

For the application standpoint I really like the idea of having several modular definitions we can use together (e.g., a time resolved definition which just brings in time resolved fields and groups). But we still have to discuss and conclude whether this will be supported in our/NIAC tools.

Reply:
> Commented on the fairmat-proposal page by Anders Frisk (Scienta Omicron) and Marek Matas (Thermofischer), that the fields `preparation_description`, `temperature` (recommended), `situation` and `gas_pressure` should be optional or recommended instead of required. The rationale is that this information is not always available or useful in the measurement context.

Please also have a look at the newly refactored NXsample. We have a new description for sample history ([NXsample_history](https://github.com/FAIRmat-NFDI/nexus_definitions/blob/fairmat/contributed_definitions/NXsample_history.nxdl.xml)) that makes use of describing [activities](https://github.com/FAIRmat-NFDI/nexus_definitions/blob/fairmat/contributed_definitions/NXactivity.nxdl.xml) and processes on the sample, in line with the base sections proposed by Area A. This should be useful for sample preparation.

We have discussed whether the sample environment (e.g. temperature and gas_pressure) should be stored on the sample level (currently possible [here](https://github.com/FAIRmat-NFDI/nexus_definitions/blob/937af0945ee1ae806ea655bc2cd81478a84d6a8e/base_classes/NXsample.nxdl.xml#L514C4-L521C15)) or on the instrument level. For me, it makes sense to have it on the instrument level and then maybe have a link towards that environment description also in the sample. In any way, these descriptions should involve NXenvironments.

Reply:
Thank you! What's your suggestion on how to implement this here? As far as I see it we can still use NXsample the same way we did before, but do you think it's better to require other fields as we currently do? Do we already have an agreement which fields we want to require for all of our appdefs?

Reply:
I don't think we should use NXsample as before, i.e., with a long flat list of fields, but rather make use of the new optional base classes NXsample_history (which includes one or more activities/processes), NXenvironment, and NXsubstance. I have already started to update the [XPS reader](https://github.com/FAIRmat-NFDI/pynxtools/blob/mpes_sample_refactor/pynxtools/dataconverter/readers/xps/config_file_sle.json) and [example](https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-remote-tools-hub/-/tree/xps_eln_update/docker/xps/example/Example%202%20-%20SPECS%20native%20SLE%20data%20format), but so far, these have not been pushed because we are also planning to refactor all the other MPES-related base classes. For using the new base classes in NXsample, I was thinking along the lines of what is given in the [ELN file](https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-remote-tools-hub/-/blob/xps_eln_update/docker/xps/example/Example%202%20-%20SPECS%20native%20SLE%20data%20format/eln_data_sle.yaml) in the new example. Any environmental conditions should be described by a NXenvironment (incl. an NXsensor) and any sample preparation step should be described by an instance of NXphysical_process in NXsample_history. Then there is also the possibility of describing individual sample components (which are grouped in a NXsample_component_set). 

We have not yet discussed which fields should be required in the appdefs. If we want to keep the same entities required as before, I recommend to restructure the AppDef such that the new base classes are used. If you agree, I can do this for NXsample in a separate PR to be merged into this one.

Reply:
This is how NXsample_component_set, NXsample_component, and NXsubstance can be used to granularize the description of sample components.

![image](https://github.com/FAIRmat-NFDI/nexus_definitions/assets/50139597/a904ba2c-0078-48d6-9e73-9fdbf3a2fa46)

Reply:
> We have not yet discussed which fields should be required in the appdefs. If we want to keep the same entities required as before, I recommend to restructure the AppDef such that the new base classes are used. If you agree, I can do this for NXsample in a separate PR to be merged into this one.

Yes, I think its good to have this to see the changes. We can still discuss whether to keep certain fields or not but I think it's good to get a feeling for the new structure. If you want you can also directly commit it to this branch. I think we can discuss it all together here.

Thank you for the nice work on NXsample. It looks really nice! So let's get it implemented everywhere :)